bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+                            currentInputHash = 0L;
+                        } catch (VizioException e) {
+                            logger.debug("Unable to set current source on the Vizio TV, source: {}, Exception: {}",
+                                    command, e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                    break;
+                case ACTIVE_APP:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            Optional<VizioApp> selectedApp = userConfigApps.stream()
+                                    .filter(a -> command.toString().equals(a.getName())).findFirst();
+
+                            if (selectedApp.isPresent()) {
+                                communicator.launchApp(selectedApp.get().getConfig());
+                            } else {
+                                logger.debug("Unknown app name: '{}', check that it exists in App List configuration",
+                                        command);
+                            }
+                        } catch (VizioException e) {
+                            logger.debug("Unable to launch app name: '{}' on the Vizio TV, Exception: {}", command,
+                                    e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                    break;
+                case CONTROL:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            handleControlCommand(command);
+                        } catch (VizioException e) {
+                            logger.debug("Unable to send control command: '{}' to the Vizio TV, Exception: {}", command,
+                                    e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                    break;
+                case BUTTON:
+                    synchronized (sequenceLock) {
+                        try {
+                            KeyCommand keyCommand = KeyCommand.valueOf(command.toString().toUpperCase(Locale.ENGLISH));
+                            communicator.sendKeyPress(keyCommand.getJson());
+                        } catch (IllegalArgumentException | VizioException e) {
+                            logger.debug("Unable to send keypress to the Vizio TV, key: {}, Exception: {}", command,
+                                    e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.